core_impl/src/main/java/com/google/instrumentation/trace/SpanImpl.java

```diff
@@ -46,6 +54,15 @@
   // The start time of the span. Set when the span is created iff the RECORD_EVENTS options is
   // set, otherwise 0.
   private final long startNanoTime;
+  // List of the recorded annotations.
+  @GuardedBy("this")
```

Comment:
Do we need this synchronization? All updates should happen in a single thread, so necessary for reads?